test/avalanche/pangolin/pangolin_exchange.test.ts

```diff
@@ -0,0 +1,283 @@
+import { expect } from "chai";
+import hre from "hardhat";
+
+const { web3, deployments, waffle, ethers } = hre;
+const { provider, deployContract } = waffle;
+
+import { deployAndEnableConnector } from "../../../scripts/tests/deployAndEnableConnector";
+import { buildDSAv2 } from "../../../scripts/tests/buildDSAv2";
+import { encodeSpells } from "../../../scripts/tests/encodeSpells";
+import { getMasterSigner } from "../../../scripts/tests/getMasterSigner";
+import { addLiquidity } from "../../../scripts/tests/addLiquidity";
+import { addresses } from "../../../scripts/tests/avalanche/addresses";
+import { abis } from "../../../scripts/constant/abis";
+import type { Signer, Contract } from "ethers";
+
+import { ConnectV2PngAvalanche__factory } from "../../../typechain";
+
+const PNG_ADDRESS  = "0x60781C2586D68229fde47564546784ab3fACA982";
+const WAVAX_ADDRESS = "0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7";
+const PNG_AVAX_LP_ADDRESS = "0xd7538cABBf8605BdE1f4901B47B8D42c61DE0367";
+
+describe("Pangolin DEX - Avalanche", function () {
+    const pangolinConnectorName = "PANGOLIN-TEST-A"
+    
+    let dsaWallet0: any;
+    let masterSigner: any;
+    let instaConnectorsV2: any;
+    let pangolinConnector: any;
+    
+    const wallets = provider.getWallets()
+    const [wallet0, wallet1, wallet2, wallet3] = wallets
+    before(async () => {
+        await hre.network.provider.request({
+            method: "hardhat_reset",
+            params: [
+                {
+                    forking: {
+                        jsonRpcUrl: `https://api.avax.network/ext/bc/C/rpc`,
+                        blockNumber: 8197390
+                    },
+                },
+            ],
+        });
+
+        masterSigner = await getMasterSigner();
+        instaConnectorsV2 = await ethers.getContractAt(
+            abis.core.connectorsV2,
+            addresses.core.connectorsV2
+        );
+
+        // Deploy and enable Pangolin Connector
+        pangolinConnector = await deployAndEnableConnector({
+            connectorName: pangolinConnectorName,
+            contractArtifact: ConnectV2PngAvalanche__factory,
+            signer: masterSigner,
+            connectors: instaConnectorsV2
+        });
+        console.log("Pangolin Connector address: "+ pangolinConnector.address);
+    })
+
+    it("Should have contracts deployed.", async function () {
+        expect(!!instaConnectorsV2.address).to.be.true;
+        expect(!!pangolinConnector.address).to.be.true;
+        expect(!!masterSigner.address).to.be.true;
+      });
+    
+    describe("DSA wallet setup", function () {
+        it("Should build DSA v2", async function () {
+            dsaWallet0 = await buildDSAv2(wallet0.address)
```

Comment:
Replace everyplace where it is used!
```suggestion
            dsaWallet0 = await buildDSAv2(wallet0.getAddress())
```